js/ui/html_editor/modules/tableResizing.js

```diff
@@ -0,0 +1,491 @@
+
+const TIMEOUT = 100;
+
+const DRAGGABLE_ELEMENT_OFFSET = 2;
+
+const MODULE_NAMESPACE = 'dxHtmlTableResizingModule';
+
+const POINTERDOWN_EVENT = addNamespace('dxpointerdown', MODULE_NAMESPACE);
+const SCROLL_EVENT = addNamespace('scroll', MODULE_NAMESPACE);
+
+export default class TableResizingModule extends BaseModule {
+    constructor(quill, options) {
+        super(quill, options);
+        this.enabled = !!options.enabled;
+        this._tableResizeFrames = [];
+        this._minColumnWidth = options.minColumnWidth ?? DEFAULT_MIN_COLUMN_WIDTH;
+        this._minRowHeight = options.minRowHeight ?? DEFAULT_MIN_COLUMN_WIDTH / 2;
+        this._quillContainer = this.editorInstance._getQuillContainer();
+        this._quillInstance = quill;
+
+        if(this.enabled) {
+            this._attachResizerTimeout = setTimeout(() => {
+
+                const $tables = this._findTables();
+                if($tables.length) {
+                    this._fixTablesWidths($tables);
+                    this._createResizeFrames($tables);
+                    this._updateFramesPositions();
+                    this._updateFramesSeparators();
+                }
+
+                this._attachEvents();
+            }, TIMEOUT);
+
+            this.addCleanCallback(this.clean.bind(this));
+            this._resizeHandler = _windowResizeCallbacks.add(this._resizeHandler.bind(this));
+        }
+    }
+
+    _attachEvents() {
+        eventsEngine.on(this.editorInstance._getContent(), SCROLL_EVENT, this._updateFramesPositions.bind(this));
+
+        this._quillTextChangeHandler = this._getQuillTextChangeHandler.bind(this);
+        this._quillInstance.on('text-change', this._quillTextChangeHandler);
+    }
+
+    _detachEvents() {
+        eventsEngine.off(this.editorInstance._getContent(), MODULE_NAMESPACE);
+        this._quillInstance.off('text-change', this._quillTextChangeHandler);
+    }
+
+    _getQuillTextChangeHandler(delta, oldContent, source) {
+        if(this._isTableChanges()) {
+            this._removeResizeFrames();
+            const $tables = this._findTables();
+
+            if(source === 'api') {
+                this._fixTablesWidths($tables);
+            }
+
+            this._updateColumnsWidth($tables);
+            this._createResizeFrames($tables);
+            this._updateFramesPositions();
+            this._updateFramesSeparators();
+        } else {
+            this._updateFramesPositions();
+        }
+    }
+
+    _resizeHandler() {
+        this._windowResizeTimeout = setTimeout(() => {
+            const $tables = this._findTables();
+            each($tables, (index, table) => {
+                const $table = $(table);
+                const actualTableWidth = $table.outerWidth();
+                const lastTableWidth = this._getWidthAttrValue($table);
+                if(Math.abs(actualTableWidth - lastTableWidth) > 1) {
+                    $table.attr('width', actualTableWidth + 'px');
```

Comment:
table `width` attribute is not supported by HTML5. Cannot we use css width instead of it? Seems like it can also make code cleaner